occurrences/models.py

```diff
@@ -427,7 +428,13 @@ class StudyGroup(TimestampedModel):
     person = models.ForeignKey(
         "organisations.Person", verbose_name=_("person"), on_delete=models.PROTECT
     )
-    name = models.CharField(max_length=1000, blank=True, verbose_name=_("name"))
+    # Tprek / Service map id for school or kindergarten from the city of Helsinki
+    unit_id = models.CharField(
+        max_length=255, verbose_name=_("unit id"), blank=True, null=True
```

Comment:
What does it mean for us if `unit_id` is `None` instead of an empty string i.e. should we just remove `null=True`?

Reply:
`None` represents the situation when the value is really not given. In this specific case it means that unit, a school or kindergarten, is outside of the city of Helsinki. The unit_id -field stores a loose id which can be linked to a TPREK service unit (in the Servicemap or the LinkedEvents). If the unit_id is not given, the school is outside from the city of Helsinki and only the name should and must be given.

I would say that from the data and database perspective, it's much more clear and therefore an important thing to store the value as NULL, when the value is unknown or not given. Me myself, I would not allow the `blank=True` here, or almost never, because an empty string really does not represent a value that is not given, but rather a value which would like to be bypassed in a form somehow. I think that setting the field to both, nullable and blankable is a preferred way in Django or at least in Palvelutarjotin, because the model file is full of these:

```
    enrolment_start = models.DateTimeField(
        verbose_name=_("enrolment start"), blank=True, null=True
    )
    # Enrolment will be close x days before the occurrence start
    enrolment_end_days = models.PositiveSmallIntegerField(
        verbose_name=_("enrolment end days"), blank=True, null=True
    )
    external_enrolment_url = models.URLField(
        verbose_name=_("enrolment url"), blank=True, null=True
    )
    needed_occurrences = models.PositiveSmallIntegerField(
        verbose_name=_("needed occurrence"), default=1
    )
```

If I'm to decide, I would set them all to `null=True, blank= False`. Is the blank-attribute here to make the validation simpler or just to fool Django somehow?

Reply:
To me it's irritating that I need to check things like this https://github.com/City-of-Helsinki/palvelutarjotin/pull/221/files#diff-32cfd933813e52d2e31fe5b92c696bb856c1d524a38666af02944ce964511930R40. Therefore I would not allow them to be blank, but only None.

Reply:
`CharField`s are a bit different from the other field types since empty string usually can be considered as empty, where as there's no such value for datetimes. Having `blank=False` (which is the default) makes sure that if there's any type of form validation happening, it will not allow an empty value on that field. I'm not sure how the validations are being run for graphene, but at least normal Django forms (including the ones in Django admin) and Django Rest framework serializers run the validations.

But I guess since there's a semantic difference here between `None` and `""` it's probably ok to have `null=True` here.